core/providers.tf

```diff
@@ -16,14 +8,14 @@ terraform {
 }
 
 provider "aws" {
-  region     = var.region
+  region     = local.region
   access_key = var.aws_access_key
   secret_key = var.aws_secret_key
```

Comment:
since you already set the variable sets as mentioned in [this comment](https://github.com/sanG-github/nimble-devops-ic-infrastructure/pull/20#discussion_r1343917138), these attributes are no longer needed
```suggestion
```

Reply:
ah, I know why it doesn't work. Your variable sets was not properly configured
<img width="938" alt="image" src="https://github.com/sanG-github/nimble-devops-ic-infrastructure/assets/1896814/52f2026d-d78a-4900-80a4-8f6aece7374d">
- They should be marked as sensitive to not showing their values on Terraform Console
- The names are incorrect, they must be `AWS_ACCESS_KEY_ID` & `AWS_SECRET_ACCESS_KEY` ([source](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#environment-variables))
- They should be set as environment variables, instead of Terraform
<img width="1000" alt="image" src="https://github.com/sanG-github/nimble-devops-ic-infrastructure/assets/1896814/8d672c68-c9f5-4803-ad2f-63e9b2e7036e">

Reply:
Thank you so much, the changes have been addressed in 5d20a7c ✅